src/helpers/transport/stream/mod.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(feature = "web-app")]
+mod axum_body;
+mod box_body;
+mod collection;
+mod input;
+
+#[cfg(feature = "web-app")]
+pub use axum_body::WrappedAxumBodyStream;
+pub use box_body::WrappedBoxBodyStream;
+pub use collection::{StreamCollection, StreamKey};
+pub use input::{LengthDelimitedStream, RecordsStream};
+
+use crate::error::BoxError;
+use bytes::Bytes;
+use futures::Stream;
+
+pub trait BytesStream: Stream<Item = Result<Bytes, BoxError>> + Send {
```

Comment:
is it worth making it generic over a reference to [u8]?

Reply:
That does seem like it could be useful, but I am inclined not to spend time on it until there is a more specific need. I'm worried that there may be places where handing off our generic version to something that strictly requires `Bytes` is problematic.